### ISSUE TITLE
Fails to download cartridge

### ISSUE THREAD

**Issue by Lineflyer** [biylda/WhereYouGo#37]
Some bug reports are coming in on the c:geo support channels, that WhereYouGo fails to download cartridges when called from c:geo.

I retested this with WhereYouGo 0.9.3 on my phone and can reproduce. It seems however not related to c;geo as it also fails when calling the cartridge URL from the website and invoking WhereYouGo from there.

I am not able to debug right now, but from the UI it seems it recognizes the CGUID correctly, can log on but fails afterwards. UI shows "Lädt herunter: Fehler" (Downloading: Error)

**Comment by SchoolGuy** [comment]
I had a similar error. It occurred when a user never visited the wherigo.com website and accepted the user agreement right before the first download.

**Comment by Lineflyer** [comment]
At this moment there are 12 users contacting us with this problem and I can reproduce it with three different user names.

@SchoolGuy Is is working for you?

**Comment by SchoolGuy** [comment]
Okay so this seems to have changed recently. And it is c:geo related as far as I can see. When I download the cartridge via Firefox on my Huawei Mate20 Pro then it is downloaded without problems. If I do it with c:geo then it fails. @Lineflyer

**Comment by SchoolGuy** [comment]
I will try with a Mate 30 Pro at home tomorrow.

**Comment by Lineflyer** [comment]
Well for me it looks unrelated to c:geo as it also fails for me coming from Chrome.

**Comment by SchoolGuy** [comment]
I also found out that this is happening with the original Geocaching App

**Comment by jan-gerard** [comment]
It happens whatever I use to load the cartridge in Whereyougo. From my browser (GC.com link or WhereIGo link), or from file, or from LocusPro. It says "connecting", "logging in", "downloading", and than "Downloading: Fout (error)".
I also noticed that I cannot change the Wherigo folder in "Settings" > "General" > "Wherigo Folder". If I press "Select" and navigate to a folder, and then try to press "OK", nothing happens. Neither does pressing "Cancel" work.
What works is to copy the gwc file manually to the "files" folder shown in the above step. Then I can play that whereigo cartridge. 
So to me it seems there is something wrong with the file storage.

**Comment by ztNFny** [comment]
https://github.com/biylda/WhereYouGo/pull/38 has the fix I've been using for the last few days

I also built a release with some other fixes in it, but of course can't distribute that one

**Comment by SchoolGuy** [comment]
I would say we have to discuss to create a fork because this repository seems to be unmaintained.

**Comment by Lineflyer** [comment]
@SchoolGuy It should IMHO not be unmaintained as AFAIR the transfer to Github was done to be able to maintain it here. I sent a mail to @biylda ...lets wait if he replies.

Even if we do a fork, this will not help in publishing an update via Google Play

**Comment by Lineflyer** [comment]
In case any of the project owners/admins is reading this:

In case there is no plan to maintain this project in future, please kindly contact me. I am sure we can help.

**Comment by Lineflyer** [comment]
BTW: To publish a new version on Google Play for sure #34 needs to be done first (as per Google requirement)

**Comment by SchoolGuy** [comment]
@Lineflyer Yes I agree. I am aware of this, I have a lot on my hands but I don't want to see Wherigos/Whereyougo die, so I will take the time to help improving it!

**Comment by Lineflyer** [comment]
Lets see and wait first if the owner of this repo answers here. I additionally tried to contact the original author to somehow get in touch with responsible persons of this repo.
IMHO this repo here is the best code base to continue from, but preferably with support of the repo owner if he is no longer interested to continue the project. If he is willing to continue but maybe only currently unreachable due to season holiday, then disregard this of course.

**Comment by Lineflyer** [comment]
I have contact with the original author now, who can possibly provide access to signing keys and Google Play is needed. So we will have a possibility to do something as needed.

However I would suggest to wait some more time, to give @biylda the chance to reply first (maybe only on vacation, or similar). The main functions of WhereYouGo are functional...at this moment only the online function fails. I will report back next week.

**Comment by ztNFny** [comment]
@Lineflyer did you mail him yet? The mail is in README.md
In case a fork is needed I already have a branch with the existing PRs merged, build tools and permissions updated. I guess it'd still need dynamic permissions to be publishable in Play Store (not sure that's already enforced)

**Comment by bekuno** [comment]
@ztNFny See also my PR #34.
I think that I can implement the permission handling. I did it some days ago for the "c:geo - contacts plugin".

**Comment by Lineflyer** [comment]
I failed to contact @biylda but I am in contact with Menion (original author).
He was able to restore access to Google Play and is in parallel now trying to contact the relevant people, who previously maintained the open source code to ask for their agreement, that a new team could take over.

Meanwhile I created a [clone into the c:geo organisation](https://github.com/cgeo/WhereYouGo) and pushed some PR commit from this repository (e.g. translations). 
@ztNFny Feel free to submit PRs to the mentioned repo already. However we will wait with any further steps until reply from Menion.

**Comment by SchoolGuy** [comment]
@Lineflyer Is there something I can help with (I don't want to work on sth which is already worked on)?

**Comment by Lineflyer** [comment]
After "timeout" of waiting for any reaction Menion enabled the "c:geo team" to get access to Google Play account and signing keys of WhereYouGo.
So I suggest to continue development in the above mentioned repo clone. Doing so, we have the full chain of repo, building, signing and publishing restored.

I have seen some already started contributing over there. All others: Feel invited to help as well to keep WhereYouGo alive.
The fact, that we have taken over the repo into our github org does not mean we have any suficient resources for further development by ourselves ;)

@biylda If you read this only at a later point of time: Please get in touch with me/us, we will of course hand it back anytime if you are still interested to maintain it.
